crypto/crypto_aesctr.c

```diff
@@ -150,6 +175,13 @@ crypto_aesctr_stream(struct crypto_aesctr * stream, const uint8_t * inbuf,
 {
 	size_t bytemod;
 
+#if defined(CPUSUPPORT_X86_AESNI)
+	if ((use_aesni == 1) && (buflen >= 16)) {
```

Comment:
Ouch, I made a terrible mistake here: we can't pick and choose to use the plain vs. aesni one.  At least, not in the current design -- the plain one will be all uninitialized.  It wouldn't be impossible to share data between the two implementations, or to have the aesni version pick and choose which parts of the structure to use.